### PR TITLE
[RESTEASY-3030] Update User Guide with Jakarta EE references

### PR DESCRIPTION
**Pull request by ronsigal**
First set of changes: User Guide.

### CODE REVIEW

File: docbook/reference/en/en-US/modules/Async_Injection.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@ public class Foo {
 package my.test.asyc.resources;
 
 import io.reactivex.Single;
-import javax.ws.rs.ext.Provider;
+import jakarta.ws.rs.ext.Provider;
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Async_Injection.xml:75]: We shouldn't change this yet.

File: docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml [hunk 2]
```diff
@@ -30,8 +30,8 @@
       </para>
    
       <programlisting>
-import javax.ws.rs.Suspend;
-import javax.ws.rs.core.AsynchronousResponse;
+import jakarta.ws.rs.Suspend;
+import jakarta.ws.rs.core.AsynchronousResponse;
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml:34]: Same with these, we should not change them yet.

File: docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml [hunk 3]
```diff
@@ -67,9 +67,9 @@ public class SimpleResource
       </para>
       <para>
          <emphasis role="bold">NOTE:</emphasis> The old RESTEasy proprietary API for async http has been deprecated and may be removed as soon as RESTEasy 3.1.
-         In particular, the RESTEasy @Suspend annotation is replaced by <classname>javax.ws.rs.container.Suspended</classname>, and
+         In particular, the RESTEasy @Suspend annotation is replaced by <classname>jakarta.ws.rs.container.Suspended</classname>, and
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml:70]: Same here too

File: docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml [hunk 4]
```diff
@@ -67,9 +67,9 @@ public class SimpleResource
       </para>
       <para>
          <emphasis role="bold">NOTE:</emphasis> The old RESTEasy proprietary API for async http has been deprecated and may be removed as soon as RESTEasy 3.1.
-         In particular, the RESTEasy @Suspend annotation is replaced by <classname>javax.ws.rs.container.Suspended</classname>, and
+         In particular, the RESTEasy @Suspend annotation is replaced by <classname>jakarta.ws.rs.container.Suspended</classname>, and
          <classname>org.jboss.resteasy.spi.AsynchronousResponse</classname> is replaced by
-         <classname>javax.ws.rs.container.AsyncResponse</classname>. Note that @Suspended does not have a value field,
+         <classname>jakarta.ws.rs.container.AsyncResponse</classname>. Note that @Suspended does not have a value field,
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml:72]: Same here too

File: docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml [hunk 5]
```diff
@@ -91,8 +91,8 @@ public class SimpleResource
           Here is an example of asynchronous processing using <code>CompletionStage</code>:
       </para>
       <programlisting>
-import javax.ws.rs.Suspend;
-import javax.ws.rs.core.AsynchronousResponse;
+import jakarta.ws.rs.Suspend;
+import jakarta.ws.rs.core.AsynchronousResponse;
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Asynchronous_HTTP.xml:95]: Same here too

File: docbook/reference/en/en-US/modules/CDI.xml [hunk 6]
```diff
@@ -82,16 +82,16 @@
...
                     it is
                     bound to the Request scope.
                 </para>
             </listitem>
             <listitem>
                 <para>
-                    If a JAX-RS Provider or
-                    <code>javax.ws.rs.Application</code>
+                    If a Jakarta REST Provider or
+                    <code>jakarta.ws.rs.Application</code>
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/CDI.xml:94]: Same here too

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 7]
```diff
@@ -295,19 +295,20 @@
...
             or security constraint you want to within your web.xml, but none of them are required:
         </para>
         <para>
             <programlisting><![CDATA[
-<web-app version="3.0" xmlns="http://java.sun.com/xml/ns/javaee"
-        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd">
+<web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:310]: This should be https://jakarta.ee/xml/ns/jakartaee/web-app_4_0.xsd

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 8]
```diff
@@ -727,9 +728,10 @@ String value = ConfigProvider.getConfig().getOptionalValue("prop_name", String.c
     </para>
     
 <programlisting>
-&lt;web-app version="3.0" xmlns="http://java.sun.com/xml/ns/javaee"
+&lt;web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"&gt;
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:733]: This should be https://jakarta.ee/xml/ns/jakartaee/web-app_4_0.xsd

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 9]
```diff
@@ -765,9 +767,10 @@ String value = ConfigProvider.getConfig().getOptionalValue("prop_name", String.c
    </para>
    
    <programlisting>
-&lt;web-app version="3.0" xmlns="http://java.sun.com/xml/ns/javaee"
+&lt;web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"&gt;
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:772]: This should be https://jakarta.ee/xml/ns/jakartaee/web-app_4_0.xsd

File: docbook/reference/en/en-US/modules/Installation_Configuration.xml [hunk 10]
```diff
@@ -804,9 +807,10 @@ String value = ConfigProvider.getConfig().getOptionalValue("prop_name", String.c
    </para>
 
 <programlisting>
-&lt;web-app version="3.0" xmlns="http://java.sun.com/xml/ns/javaee"
+&lt;web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"&gt;
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/Installation_Configuration.xml:812]: This should be https://jakarta.ee/xml/ns/jakartaee/web-app_4_0.xsd

File: docbook/reference/en/en-US/modules/RESTEasy_Spring_Integration.xml [hunk 11]
```diff
@@ -153,9 +153,10 @@
             as importing the <code>springmvc-resteasy.xml</code> file into your base Spring beans xml file. Here's an
             example <code>web.xml</code> file:
 
-            <programlisting><![CDATA[<web-app version="3.0" xmlns="http://java.sun.com/xml/ns/javaee"
+            <programlisting><![CDATA[<web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd">
+         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
```
Comment (jamezp) [inline docbook/reference/en/en-US/modules/RESTEasy_Spring_Integration.xml:158]: This should be https://jakarta.ee/xml/ns/jakartaee/web-app_4_0.xsd

File: docbook/reference/en/en-US/modules/Async_Injection.xml (hunk 1 above)
Comment (ronsigal) [reply docbook/reference/en/en-US/modules/Async_Injection.xml:75]: Hey @jamezp,

Actually, I realized that I'm probably working on the wrong branch. Doesn't it make sense to make changes like the above on a branch where we really are on Jakarta REST 3.0+? If the code is referring to javax.ws.rs.*, then what's the point of making any of these changes? 

Looking back at Alessio's agenda comment of September 8, he said, "Nov 16th is the hard deadline for removing any JavaEE compliance statement which is not explicitly referring to a past RESTEasy version." So, I think I've overreacted. Basically, everything I've done should go on a RESTEasy 6.0 branch, which we don't have. Can I take ee10 and merge main into it?

Anyway, it seems to me that for now, I should just look for any compliance statements.

-Ron

**Pull request closed without merging**